data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -121,46 +131,110 @@ public void start(final Promise<Void> startPromise) {
+          .<Void>mapEmpty()
+          .onComplete(startPromise);
     } else {
-      this.httpServer.requestHandler(handler)
+      this.httpServer
+          .requestHandler(handler)
           .exceptionHandler(startPromise::tryFail)
           .listen(this.httpServerOptions.getPort(), this.httpServerOptions.getHost())
-          .<Void>mapEmpty().onComplete(startPromise);
+          .<Void>mapEmpty()
+          .onComplete(startPromise);
     }
+    vertx.<Watch>executeBlocking(
+        promise -> {
+          Watch watch =
+              client
+                  .secrets()
+                  .inNamespace("knative-eventing")
+                  .withName("receiver-tls-secret")
+                  .watch(
+                      new Watcher<Secret>() {
+
+                        @Override
+                        public void eventReceived(Action action, Secret secret) {
+                          // This block will be called when the Secret is added, modified, or
+                          // deleted.
+
+                          // Update SSL configuration by using updateSSLOptions
+                          PemKeyCertOptions keyCertOptions =
+                              new PemKeyCertOptions()
+                                  .setKeyPath(TLS_KEY_FILE_PATH)
+                                  .setCertPath(TLS_CRT_FILE_PATH);
+
+                          httpServer.updateSSLOptions(
+                              new SSLOptions().setKeyCertOptions(keyCertOptions));
+
+                          // Restart server
+                          ReceiverVerticle.this.httpsServer.close();
+                          ReceiverVerticle.this.httpsServer =
+                              vertx.createHttpServer(ReceiverVerticle.this.httpsServerOptions);
+                          ReceiverVerticle.this
+                              .httpsServer
+                              .requestHandler(handler)
+                              .exceptionHandler(startPromise::tryFail)
+                              .listen(
+                                  ReceiverVerticle.this.httpsServerOptions.getPort(),
+                                  ReceiverVerticle.this.httpsServerOptions.getHost());
+                        }
+
+                        @Override
+                        public void onClose(WatcherException cause) {
+                          if (cause != null) {
+                            logger.error("Watcher onClose with exception", cause);
+                          }
+                        }
+                      });
+
+          promise.complete(watch);
+        },
+        res -> {
+          if (res.succeeded()) {
+            Watch watch = res.result();
+            // Store this `watch` reference somewhere in order to be able to close it in the future
+          } else {
+            logger.error("Watcher exception", res.cause());
+          }
+        });
```

Comment:
We should watch the filesystem instead of watching secret directly with Kube clients